sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -490,6 +529,56 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
             });
     }
 
+    /**
+     * Gets token from shared token cache
+     * */
+    public Mono<AccessToken> authenticateWithSharedTokenCache(TokenRequestContext request, String username) {
+        // find if the Public Client app with the requested username exists
+        return Mono.fromFuture(() -> getPublicClientApplication(true).getAccounts())
+                .onErrorMap(t -> new CredentialUnavailableException(
+                        "Cannot get accounts from token cache. Error: " + t.getMessage(), t))
+                .flatMap(set -> {
+                    IAccount requestedAccount;
+                    Map<String, IAccount> accounts = new HashMap<>(); // home account id -> account
+
+                    for (IAccount cached : set) {
+                        if (username == null || username.equals(cached.username())) {
+                            if (!accounts.containsKey(cached.homeAccountId())) { // only put the first one
+                                accounts.put(cached.homeAccountId(), cached);
+                            }
+                        }
+                    }
+
+                    if (set.size() == 0) {
+                        return Mono.error(new CredentialUnavailableException("SharedTokenCacheCredential "
```

Comment:
Should we use `return FluxUtils.monoError(logger, new ....)`? Or is it preferable not to log it

Reply:
We talked to Jonathan and since this IdentityClient isn't used directly by customers and we have downstream credentials to do the logging - we shouldn't log here.